File: CCC-school-work/comporg/pnc2/bstrong2/java/primeregbs.java
import java.text.DecimalFormat;
/////////////////////////////////////////////////////////////
//
// Public class in java, java looks for the main class when
// compliling. Then have the function that is like the same
// as the main function in c.
//
public class primeregbs
{

	public static void main(String[] args)
	{
		/////////////////////////////////////////////////////
		//
		// Declaring the variables.
		//
		int dividend = 2, foundPrimes = 0;
		int flag,maxPrimes, lowerbound = 2;
		double divisorMax,store;
		long start, end,time;
		String tmp;
		
		/////////////////////////////////////////////////////
		//
		// Then check to see if the length is 0, if it is
		// then print out an error and then exit the program
		// Then check to see if the amount of arguments is
		// more than what the program can accept, if it is
		// then print off an error message and exit the
		// program.
		//
		if(args.length == 0)
		{
			System.err.printf("Enter Something valid\n");
			System.exit(0);
		}
		else if(args.length > 1)
		{
			System.err.printf("Too many arguments\n");
			System.exit(1);
		}
		/////////////////////////////////////////////////////
		//
		// Now set a temp variable = to the number that the
		// user specified. Set max primes to the integer that
		// the user input. Then start the calculation timer.
		//
		tmp = args[0];
		maxPrimes = Integer.parseInt(tmp);
		start = System.nanoTime();
		/////////////////////////////////////////////////////
		//
		// This for loops goes until the amount of primes has
		// been found. It increments the lowerbound after 
		// every iteration. After every iteratom set divident
		// = to 2, plag = to 0, then do the calculation to
		// find the divisormax
		//
		for(; foundPrimes < maxPrimes; lowerbound++)
		{
			dividend = 2;
			flag = 0;	
			divisorMax = Math.sqrt(lowerbound);
			/////////////////////////////////////////////////
			//
			// Now have another for loop that goes as many
			// times as the divisorMax. In the loop there
			// is an if statement that checks to see if there
			// are any remainders when dividing. If there
			// is then set a flag = to 1 and break out of the
			// loop. Increase dividend after every loop.
			//
			for(; dividend <= divisorMax;dividend++)
			{
				if(lowerbound % dividend == 0)
				{
					flag = 1;
					break;
				}
			}
			/////////////////////////////////////////////////
			//
			// Now check to see if flag is = to 0, if it is
			// then print out that number because it's prime
			// then increment foundprimes by 1.
			//
			if(flag == 0)
			{
				System.err.printf("%d ", lowerbound);
				foundPrimes++;
			}
		}	
		/////////////////////////////////////////////////////
		//
		// Then we do the final time calculation, print off
		// a new line. Then end the calculation for time.
		// Then take a time variable and get the total time
		// that the program ran for in nano time. Then 
		// divide accordingly to get the correct values. Then
		// make sure that the values are limited by 4 decimal
		// places. Print out a tap then the actual time.
		//
		System.err.printf("\n");
		end = System.nanoTime();
		time = end - start;
		store = time/1000000;	
		store = store/1000;
		DecimalFormat numberFormat = new DecimalFormat("#.####");
		System.out.print("	");
		System.out.println(numberFormat.format(store));
	}
}

File: CCC-school-work/comporg/pnc2/bstrong2/js/primeregbs.js
/////////////////////////////////////////////////////////
//
// So here we are with another program yet another 
// primeregbs :D. Alright so first thing is first and
// it's needed to check to see if there are enough 
// arguments, if not then print an error to the user.
// Also give an error if the user inputs anthing less
// than or = to 0.
//
if(process.argv.length < 3)
	process.stderr.write("Invalid amount of arguments\n");
else if(process.argv[2] <= 0)
	process.stderr.write("Don't you want to see primes?\n");
/////////////////////////////////////////////////////////
//
// Next it's time to declare all of the used variables.
// Make max = to the number of primes the user wants
// displayed.
//
var divisor, lowerbound = 2, max,maxFound = 0, flag;
var square,start, end, time, secondsS, secondsE,secondsT;
max = process.argv[2];
/////////////////////////////////////////////////////////
//
// This is to start displaying the time. So it was a big
// issue to display the time xD. So i get the mill and
// seconds seperatly because if the mill exceeded 1000
// then the number would reset.
//
// Then it's time to start the while loop, which tbh
// it's the same condition that I used in the C program.
// To keep looping until the amount of primes has 
// been found.
//
start = new Date().getMilliseconds();
secondsS = new Date().getSeconds();
while(maxFound < max)
{
	/////////////////////////////////////////////////////
	//
	// Now for inside the loop. Set divisor to 2 at the
	// start of every iteration, set flag to 0 and find
	// the squareroot of the number being tested. Then
	// have a while loop to to go as many times as the
	// square root, inside the loop have an if checking
	// for a remainder, if a number with no remainder
	// is found then set off flag and end loop.
	//
	divisor = 2;
	flag = 0;
	square = Math.sqrt(lowerbound);
	while(divisor <= square)
	{
		if(lowerbound % divisor == 0)
		{
			flag = 1;
			break;
		}
		divisor++;
	}
	/////////////////////////////////////////////////////
	//
	// Have an if to check the flag(if the number is 
	// prime) then write to stderr the number. increment
	// values as needed(explained it many times already
	// xD).
	//
	if(flag == 0)
	{
		process.stderr.write(lowerbound + " ");
		maxFound++;	
	}
	lowerbound++;
}
/////////////////////////////////////////////////////////
//
// Now that all of the processing is done time to cut
// the timers. Set a var time = to the end time - the
// start time. Then do the same thing for the seconds
// time. (In class you said that you used Date().getTime()
// I don't know how you got arround the milliseconds thing)
//
// Next take the milliseconds time and divide it so it
// is in 4 decimal places. Then add the total amount of
// seconds the program ran for. Print that value.
//
end = new Date().getMilliseconds();
secondsE = new Date().getSeconds();
time = end - start;
secondsT = secondsE - secondsS;

time /= 10000;
time += (secondsT);
console.log("\n	" + (time));
